source/ambassador_host_test.go

```diff
@@ -17,26 +17,562 @@ limitations under the License.
+				},
+			},
+		},
+		{
+			title:            "valid non-matching annotation filter label",
+			targetNamespace:  "",
+			annotationFilter: "kubernetes.io/ingress.class=external-ingress",
+			labelSelector: labels.Everything(),
+			loadBalancer: fakeAmbassadorLoadBalancerService{
+				ips:       []string{"8.8.8.8"},
+				hostnames: []string{"lb.com"},
+				name:      "emissary",
+				namespace: "emissary-ingress",
+			},
+			ambassadorHostItems: []fakeAmbassadorHost{
+				{
+					name:      "fake1",
+					namespace: "testing1",
+					hostname:  "fake1.org",
+					annotations: map[string]string{
+						"external-dns.ambassador-service": "emissary-ingress/emissary",
+						"kubernetes.io/ingress.class":     "internal-ingress",
+					},
+				},
+			},
+			expected: []*endpoint.Endpoint{},
+		},
+		{
+			title:            "valid matching label filter expression",
+			targetNamespace:  "",
+			// annotationFilter: "kubernetes.io/ingress.class in (external-ingress)",
+			labelSelector: labels.SelectorFromSet(labels.Set{"kubernetes.io/ingress.class": "external-ingress"}),
+			loadBalancer: fakeAmbassadorLoadBalancerService{
+				ips:       []string{"8.8.8.8"},
+				hostnames: []string{"lb.com"},
+				name:      "emissary",
+				namespace: "emissary-ingress",
+			},
+			ambassadorHostItems: []fakeAmbassadorHost{
+				{
+					name:      "fake1",
+					namespace: "testing1",
+					hostname:  "fake1.org",
+					annotations: map[string]string{
+						"external-dns.ambassador-service": "emissary-ingress/emissary",
+						"kubernetes.io/ingress.class":     "external-ingress",
+					},
+					labels: map[string]string{
+						"kubernetes.io/ingress.class": "external-ingress",
+					},
+				},
+			},
+			expected: []*endpoint.Endpoint{
+				{
+					DNSName: "fake1.org",
+					Targets: endpoint.Targets{"8.8.8.8"},
+				},
+				{
+					DNSName: "fake1.org",
+					Targets: endpoint.Targets{"lb.com"},
+				},
+			},
+		},
+		{
+			title:            "valid non-matching label filter expression",
+			targetNamespace:  "",
+			// annotationFilter: "kubernetes.io/ingress.class in (external-ingress)",
+			labelSelector: labels.SelectorFromSet(labels.Set{"kubernetes.io/ingress.class": "external-ingress"}),
+			loadBalancer: fakeAmbassadorLoadBalancerService{
+				ips:       []string{"8.8.8.8"},
+				hostnames: []string{"lb.com"},
+				name:      "emissary",
+				namespace: "emissary-ingress",
+			},
+			ambassadorHostItems: []fakeAmbassadorHost{
+				{
+					name:      "fake1",
+					namespace: "testing1",
+					hostname:  "fake1.org",
+					annotations: map[string]string{
+						"external-dns.ambassador-service": "emissary-ingress/emissary",
+						"kubernetes.io/ingress.class":     "internal-ingress",
+					},
+					labels: map[string]string{
+						"kubernetes.io/ingress.class": "internal-ingress",
+					},
+				},
+			},
+			expected: []*endpoint.Endpoint{},
+		},
+		{
+			title:            "valid matching label filter expression for single host",
```

Comment:
This test case seems to be a "superset" of `valid matching label filter expression` one, so why not keeping only this one?

Reply:
I have now removed `valid matching label filter expression` as you are correct already matching valid label within the `valid matching label filter expression for single host` test. 

Sorry I was trying to match tests like what was done within the annotations without really thinking about it.